src/us/kbase/auth2/lib/Password.java

```diff
@@ -42,4 +61,37 @@ public void clear() {
 			password[i] = '0';
 		}
 	}
+	
+	/** Check for password validity (length and strength requirements)
+	 * @throws IllegalPasswordException if the password is not valid
+	 */
+	public void checkValidity() throws IllegalPasswordException {
+		// check length requirements
+		if(password.length > MAX_PASSWORD_LENGTH) {
+			throw new IllegalPasswordException("Password exceeds max length ("+
+												MAX_PASSWORD_LENGTH+")");
+		}
+		
+		// check strength requirement
+		Strength strength = new Zxcvbn().measure(new String(password));
```

Comment:
Nope, no way to handle it afaik. What you're doing now is the best we can hope for, as when zx... is done with the string its reference will be released and it can be GC'd.